Error when extending layer from GitHub with `install: true`

### Environment

**Shared for Layer and Main App**
- Operating System: Windows_NT
- Node Version:     v20.11.1
- Nuxt Version:     3.11.1
- CLI Version:      3.10.1
- Nitro Version:    2.9.4
- Package Manager:  pnpm@8.15.5
- Builder:          -
- Build Modules:   -

**Layer**
- User Config:      modules, devtools
- Runtime Modules:  @nuxt/ui@2.14.2

**Main App**
- User Config:      extends, devtools
- Runtime Modules:  -

### Reproduction

Layer: https://github.com/MuhammadM1998/nuxt-layers-repro
Main App: https://github.com/MuhammadM1998/nuxt-layers-repro-main-app

### Describe the bug

**Reproduction**
- Clone the main app (Fails to fetch the layer in stackblitz don't know why)
- Run `pnpm i`

**Expected Behavior**
- It install the main app & the layer dependencies correctly.

**Actual Behavior**
- It install the main app dependencies but throws the following error when installing the layer dependencies
```shell
ERROR  Error while requiring module @nuxt/ui: Error: Cannot find module 'C:/nuxt-layers-repro-main-app/@nuxt/ui'
Require stack:
- C:\nuxt-layers-repro-main-app\index.js

 ERROR  Cannot find module 'C:/nuxt-layers-repro-main-app/@nuxt/ui'
Require stack:
- C:\nuxt-layers-repro-main-app\index.js

  Require stack:
  - index.js
  at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
  at Function.resolve (node:internal/modules/helpers:187:19)
  at Function._resolve [as resolve] (node_modules\.pnpm\jiti@1.21.0\node_modules\jiti\dist\jiti.js:1:251148)
  at resolveModule (/C:/nuxt-layers-repro-main-app/node_modules/.pnpm/@nuxt+kit@3.11.1/node_modules/@nuxt/kit/dist/index.mjs:2210:29)
  at requireModule (/C:/nuxt-layers-repro-main-app/node_modules/.pnpm/@nuxt+kit@3.11.1/node_modules/@nuxt/kit/dist/index.mjs:2215:24)
  at loadNuxtModuleInstance (/C:/nuxt-layers-repro-main-app/node_modules/.pnpm/@nuxt+kit@3.11.1/node_modules/@nuxt/kit/dist/index.mjs:2451:92)
  at async installModule (/C:/nuxt-layers-repro-main-app/node_modules/.pnpm/@nuxt+kit@3.11.1/node_modules/@nuxt/kit/dist/index.mjs:2407:47)
  at async initNuxt (/C:nuxt-layers-repro-main-app/node_modules/.pnpm/nuxt@3.11.1_@unocss+reset@0.58.6_floating-vue@5.2.2_unocss@0.58.6_vite@5.2.6/node_modules/nuxt/dist/index.mjs:4171:7)
  at async loadNuxt (/C:/nuxt-layers-repro-main-app/node_modules/.pnpm/nuxt@3.11.1_@unocss+reset@0.58.6_floating-vue@5.2.2_unocss@0.58.6_vite@5.2.6/node_modules/nuxt/dist/index.mjs:4269:5)
  at async loadNuxt (/C:/nuxt-layers-repro-main-app/node_modules/.pnpm/@nuxt+kit@3.11.1/node_modules/@nuxt/kit/dist/index.mjs:2570:19)
```

### Additional context

_No response_

### Logs

_No response_

--- a/packages/nuxt/src/core/nuxt.ts
+++ b/packages/nuxt/src/core/nuxt.ts
@@ -272,6 +272,9 @@ async function initNuxt (nuxt: Nuxt) {
     ...nuxt.options._layers.filter(i => i.cwd.includes('node_modules')).map(i => i.cwd as string),
   )
 
+  // Ensure we can resolve dependencies within layers
+  nuxt.options.modulesDir.push(...nuxt.options._layers.map(l => resolve(l.cwd, 'node_modules')))
+
   // Init user modules
   await nuxt.callHook('modules:before')
   const modulesToInstall = []

